nixos/modules/services/web-apps/keycloak.nix

```diff
@@ -544,7 +544,12 @@ in
             create_role="$(mktemp)"
             trap 'rm -f "$create_role"' EXIT
 
+            # Read the password from the credentials directory and
+            # escape any single quotes by adding additional single
+            # quotes after them.
             db_password="$(<"$CREDENTIALS_DIRECTORY/db_password")"
+            db_password="''${db_password//\'/\'\'}"
+
```

Comment:
Why is this necessary?

Reply:
It's to escape any single quotes in the password which would otherwise break the quoting of the password in the SQL snippet further down. Single quotes can be escaped in SQL by adding an additional single quote in front of them, so that's what this does.

Reply:
Ahh I see, didn't read that far, makes sense! Ideally we wouldn't have to do this escaping ourselves, but from some research it looks like there's no other way for postgresql. I think we should at least link to [postgresql's string escaping rules](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-CONSTANTS) though. Also check out that caution ssign there, it seems to imply that the string escaping can change depending on the [standard_conforming_strings](https://www.postgresql.org/docs/current/runtime-config-compatible.html#GUC-STANDARD-CONFORMING-STRINGS) parameter, whose default changed in the past.. This could mess the escaping here up :neutral_face:

Reply:
Thank you for the implementation, I did not have time so far to continue with the issue

Reply:
Okay, I've added links in the comments to the relevant pages and also an assertion that `standard_conforming_strings` are not disabled.

I also found out that MySQL allows escape sequences by default and that has to be turned off temporarily for the expected behavior. And, when testing for that, I found that escape sequences are also interpreted somewhere in Keycloak or, more likely, the JDBC driver and have to be escaped in the config file.. *sigh*

@red-hood No problem! Thank you for raising the issue! :)